lib_eio/unix/eio_unix.ml

```diff
@@ -75,3 +75,18 @@ let getnameinfo (sockaddr : Eio.Net.Sockaddr.t) =
   run_in_systhread (fun () ->
     let Unix.{ni_hostname; ni_service} = Unix.getnameinfo sockaddr options in
     (ni_hostname, ni_service))
+
+module Spawn = struct
+  let resolve_program ~paths prog =
+    if not (Filename.is_relative prog) then Some prog
```

Comment:
This condition seems to be missing relative paths that contain a `/`.
(cf. the difference between `prog` and `./prog` in posix shells)

Reply:
If that helps I came up with this:

```ocaml
let resolve_program prog =
  (* Cf. https://github.com/ocaml-multicore/eio/pull/435/files *)
  let open Caml in
  let paths =
    Option.map (fun v -> String.split_on_char ':' v) (Sys.getenv_opt "PATH")
    |> Option.value ~default:[ "/usr/bin"; "/usr/local/bin" ]
  in
  let valid p = Sys.file_exists p && not (Sys.is_directory p) in
  if String.exists (function '/' -> true | _ -> false) prog then
    if valid prog then Some prog else None
  else
    let rec loop = function
      | path :: rest ->
          let p = Filename.concat path prog in
          if valid p then Some p else loop rest
      | [] -> None
    in
    loop paths

let%expect_test _ =
  let tmpdir = Caml.Filename.get_temp_dir_name () in
  let resolve_program_and_print p =
    let clean s =
      s
      |> String.substr_replace_all ~pattern:tmpdir ~with_:"${TMPDIR}"
      |> String.substr_replace_all ~pattern:"/usr/bin" ~with_:"${sysbin}"
      |> String.substr_replace_all ~pattern:"/bin" ~with_:"${sysbin}"
    in
    Fmt.pr "%s@ -> %a\n" (clean p)
      Fmt.Dump.(option string)
      (resolve_program p |> Option.map ~f:clean)
  in
  let ( // ) = Caml.Filename.concat in
  let cmd s =
    match Caml.Sys.command s with
    | 0 -> ()
    | n -> Fmt.failwith "%S -> %d" s n
  in
  resolve_program_and_print "echo";
  [%expect {|
    echo
    -> Some "${sysbin}/echo" |}];
  resolve_program_and_print "echooooo";
  [%expect {|
    echooooo
    -> None |}];
  let abs = tmpdir // "resolve_program_and_print.sh" in
  Fmt.kstr cmd "echo bouh > %s" abs;
  resolve_program_and_print abs;
  [%expect
    {|
    ${TMPDIR}/resolve_program_and_print.sh
    -> Some "${TMPDIR}/resolve_program_and_print.sh" |}];
  let rel = "./resolve_program_and_print.sh" in
  Fmt.kstr cmd "echo bah > %s" rel;
  resolve_program_and_print rel;
  [%expect
    {|
    ./resolve_program_and_print.sh
    -> Some "./resolve_program_and_print.sh" |}];
  resolve_program_and_print (Caml.Filename.basename rel);
  [%expect {|
    resolve_program_and_print.sh
    -> None |}];
  Fmt.kstr cmd "rm -f %s %s" abs rel;
  ()
```

Reply:
`Filename.is_implicit` is likely what we want here.